lambdas/accesstoken/src/main/java/uk/gov/di/ipv/cri/address/api/handler/AccessTokenHandler.java

```diff
@@ -0,0 +1,115 @@
+import uk.gov.di.ipv.cri.address.library.helpers.ApiGatewayResponseGenerator;
+import uk.gov.di.ipv.cri.address.library.persistence.item.AddressSessionItem;
+import uk.gov.di.ipv.cri.address.library.service.AddressSessionService;
+import uk.gov.di.ipv.cri.address.library.validation.ValidationResult;
+
+import java.net.URI;
+
+public class AccessTokenHandler
+        implements RequestHandler<APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent> {
+
+    private static final Logger LOGGER = LogManager.getLogger();
+    private final AddressSessionService addressSessionService;
+
+    public AccessTokenHandler(AddressSessionService addressSessionService) {
+        this.addressSessionService = addressSessionService;
+    }
+
+    public AccessTokenHandler() {
+        this.addressSessionService = new AddressSessionService();
+    }
+
+    public APIGatewayProxyResponseEvent handleRequest(
+            APIGatewayProxyRequestEvent input, Context context) {
+        try {
+            TokenRequest tokenRequest = createTokenRequest(input.getBody());
+
+            ValidationResult<ErrorObject> validationResult =
+                    addressSessionService.validateTokenRequest(tokenRequest);
+
+            if (!validationResult.isValid()) {
+                LOGGER.error(
+                        "Invalid access token request, error description: {}",
+                        validationResult.getError().getDescription());
+                return ApiGatewayResponseGenerator.proxyJsonResponse(
+                        getHttpStatusCodeForErrorResponse(validationResult.getError()),
+                        validationResult.getError().toJSONObject());
+            }
+
+            String authorizationCodeFromRequest =
+                    ((AuthorizationCodeGrant) tokenRequest.getAuthorizationGrant())
+                            .getAuthorizationCode()
+                            .getValue();
+
+            AddressSessionItem addressSessionItem =
+                    addressSessionService.getAddressSessionItemByAuthorizationCode(
+                            authorizationCodeFromRequest);
+
+            if (addressSessionItem == null) {
+                LOGGER.error("Session with could not be found in the database.");
+                return ApiGatewayResponseGenerator.proxyJsonResponse(
+                        OAuth2Error.INVALID_GRANT.getHTTPStatusCode(),
+                        OAuth2Error.INVALID_GRANT.toJSONObject());
+            }
+            if (!authorizationCodeFromRequest.equals(addressSessionItem.getAuthorizationCode())) {
+                LOGGER.error(
+                        "Access Token could not be issued. The supplied authorization code was not found in the database.");
+                return ApiGatewayResponseGenerator.proxyJsonResponse(
+                        OAuth2Error.INVALID_GRANT.getHTTPStatusCode(),
+                        OAuth2Error.INVALID_GRANT.toJSONObject());
+            }
+
```

Comment:
We should validate the request_uri is identical to that supplied in the authorization request. We store this value. See the redirect_uri validation requirements at https://datatracker.ietf.org/doc/html/rfc6749#section-4.1.3